cmd/horcrux/cmd/config.go

```diff
@@ -0,0 +1,274 @@
+package cmd
+
+import (
+	"fmt"
+	"github.com/mitchellh/go-homedir"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+
+	"github.com/jackzampolin/horcrux/internal/signer"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+)
+
+func init() {
+	// TODO: config nodes add/remove
+	// TODO: config peers add/remove
+	// TODO: config chain-id set
+	configCmd.AddCommand(initCmd())
+	rootCmd.AddCommand(configCmd)
+}
+
+var configCmd = &cobra.Command{
+	Use:   "config",
+	Short: "Commands to configure the horcrux signer",
+}
+
+func addPeerCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "init [chain-id] [chain-nodes]",
+		Aliases: []string{"i"},
+		Short:   "initalize configuration file and home directory if one doesn't already exist",
+		Long: "initalize configuration file, use flags for cosigner configuration.\n\n" +
+			"[chain-id] is the chain id of the chain to validate\n" +
+			"[chain-nodes] is a comma seperated array of chain node addresses i.e.\n" +
+			"tcp://chain-node-1:1234,tcp://chain-node-2:1234",
+		Args: cobra.RangeArgs(1, 2),
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			return nil
+		},
+	}
+	return cmd
+}
+
+func initCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "init [chain-id] [chain-nodes]",
+		Aliases: []string{"i"},
+		Short:   "initalize configuration file and home directory if one doesn't already exist",
+		Long: "initalize configuration file, use flags for cosigner configuration.\n\n" +
+			"[chain-id] is the chain id of the chain to validate\n" +
+			"[chain-nodes] is a comma seperated array of chain node addresses i.e.\n" +
+			"tcp://chain-node-1:1234,tcp://chain-node-2:1234",
+		Args: cobra.RangeArgs(1, 2),
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			cid := args[0]
+			var cn []ChainNode
+			if len(args) == 2 {
+				cn, err = chainNodesFromArg(args[1])
+				if err != nil {
+					return err
+				}
+			}
+
+			var homeDir string
+			if cfgFile != "" {
+				homeDir = cfgFile
+			} else {
+				home, _ := homedir.Dir()
+				homeDir = path.Join(home, ".horcrux")
+			}
+
+			//TODO double check that omitting directory check does not introduce unexpected behavior
+			//if _, err := os.Stat(homeDir); !os.IsNotExist(err) {
+			//	return fmt.Errorf("%s is not empty, check for existing configuration and clear path before trying again", homeDir)
+			//}
```

Comment:
Lets re-add this? I think we handle this in root.go

Reply:
I omitted these lines while fighting the permissions issues I was running into while accessing files created by Docker containers from Go. Now that containers are run from the user, instead of root, the fix for this is to not create the config directories when the signer structs are created and to let cmd/config handle it again